src/solver/modeler/api/include/antares/solver/modeler/api/linearProblemData.h

```diff
@@ -28,20 +28,33 @@
 namespace Antares::Solver::Modeler::Api
 {
 
-class LinearProblemData
+class ILinearProblemData
+{
+    virtual double getData(std::string idTimeSeriesSet,
+                           std::string scenarioGroup,
+                           unsigned int scenario,
+                           unsigned int hour) = 0;
+};
+
+class LinearProblemData : public ILinearProblemData
 {
 public:
-    unsigned getTimeResolutionInMinutes();
-    bool hasScalarData(const std::string& key);
-    double getScalarData(const std::string& key, unsigned scenario);
-    bool hasTimedData(const std::string& key);
-    const std::vector<double>& getTimedData(const std::string& key, unsigned scenario);
-
-private:
-    std::vector<int> timeStamps_;
-    unsigned timeResolutionInMinutes_;
-    std::map<std::string, std::vector<double>> scalarData_;
-    std::map<std::string, std::vector<std::vector<double>>> timedData_;
+    double getData(std::string idTimeSeriesSet,
+                   std::string scenarioGroup,
+                   unsigned int scenario,
+                   unsigned int hour) override;
+
+//    unsigned getTimeResolutionInMinutes();
+//    bool hasScalarData(const std::string& key);
+//    double getScalarData(const std::string& key, unsigned scenario);
+//    bool hasTimedData(const std::string& key);
+//    const std::vector<double>& getTimedData(const std::string& key, unsigned scenario);
+
+//private:
+//    std::vector<int> timeStamps_;
+//    unsigned timeResolutionInMinutes_;
+//    std::map<std::string, std::vector<double>> scalarData_;
+//    std::map<std::string, std::vector<std::vector<double>>> timedData_;
```

Comment:
This data are now commented out.
It will be uncommented if we needed eventually.